### ISSUE TITLE
Model convergence

### ISSUE THREAD

**Issue by Charlyo** [HaoIrving/RefineDet-plus-plus.PyTorch#1]
Hi @HaoIrving and thank you for this implementation. I'm trying to train this model in just one cpu, but I'm not able to make it converge. This happens for both the vgg version and the resnet50 one. On you case, in which epoch do you think the model started to converge?

Also, right now, by freezing the resnet50 backone also I get nan's in the loss.

Which parameters do you think could be feasible for just a 1 gpu training?

Thank you for your time!

**Comment by HaoIrving** [comment]
Hi @Charlyo Thanks for your attention to this repository. In my experience, the loss usually started to converge after [10 warm up epoches](https://github.com/HaoIrving/RefineDet-plus-plus.PyTorch/blob/master/train_refinedet.py#L298), the loss value at here was about 10 to 13 (I can't remember exactly)

For vgg case, the hyper-parameters provided are `--lr 4e-3 --ngpu 4 --batch_size 32`, you can try `--lr 1e-3 --ngpu 1 --batch_size 8` for single gpu training ( learning rate and batchsize are essential for the convergence, remember to tune them). In my experience, the vgg+bn backbone is easier to converge than the original vgg backbone. If not converged, tuning the learning rate and batchsize or restarting the training to find better initial model parameters may solve it. 

For resnet case, I hadn't find the solution to make it converge, sorry for that. Please remind me if you find the solution, I will apreciate that, thanks.

**Comment by HaoIrving** [comment]
Hi @Charlyo, when converged, the final loss is about 2 to 5.

**Comment by Charlyo** [comment]
@HaoIrving Could you try to use gradient clipping to avoid any problems with convergeance? 

https://pytorch.org/docs/stable/generated/torch.nn.utils.clip_grad_norm_.html

Max value around 35 should work.

**Comment by HaoIrving** [comment]
Thanks for your advice, I will have a try when free. Besides, for the incompleteness and insignificance of the code's features, I want to close this repository now. Anyway, thank you for your attention to this repository.
